frontend/packages/kubevirt-plugin/src/components/modals/scheduling-modals/tolerations-modal/tolerations-modal.tsx

```diff
@@ -0,0 +1,166 @@
+} from '@console/internal/components/utils';
+import { k8sPatch, Toleration, TaintEffect, K8sResourceKind } from '@console/internal/module/k8s';
+import { NodeModel } from '@console/internal/models';
+import { isLoaded, getLoadedData, getLoadError } from '../../../../utils';
+import { ModalFooter } from '../../modal/modal-footer';
+import { VMLikeEntityKind } from '../../../../types/vmLike';
+import { getVMLikeModel, getTolerations } from '../../../../selectors/vm';
+import { getTolerationsPatch } from '../../../../k8s/patches/vm/vm-scheduling-patches';
+import { NodeChecker } from '../shared/NodeChecker/node-checker';
+import { useNodeQualifier } from '../shared/hooks';
+import { LabelsList } from '../../../LabelsList/labels-list';
+import { TOLERATIONS_MODAL_TITLE, TOLERATIONS_EFFECTS } from '../shared/consts';
+import { useIDEntities } from '../../../../hooks/use-id-entities';
+import { useCollisionChecker } from '../../../../hooks/use-collision-checker';
+import { IDLabel } from '../../../LabelsList/types';
+
+export const TModal = withHandlePromise(
+  ({
+    nodes,
+    close,
+    handlePromise,
+    inProgress,
+    errorMessage,
+    vmLikeEntity,
+    vmLikeEntityLoading,
+  }: TModalProps) => {
+    const vmLikeFinal = getLoadedData(vmLikeEntityLoading, vmLikeEntity);
+    const loadError = getLoadError(nodes, NodeModel);
+
+    const [
+      tolerationsLabels,
+      setTolerationsLabels,
+      onLabelAdd,
+      onLabelChange,
+      onLabelDelete,
+    ] = useIDEntities<TolerationLabel>(
+      getTolerations(vmLikeEntity)?.map((toleration, id) => ({ ...toleration, id })) || [],
+    );
+
+    const qualifiedNodes = useNodeQualifier(tolerationsLabels, nodes);
+
+    const [showCollisionAlert, reload] = useCollisionChecker<VMLikeEntityKind>(
+      vmLikeFinal,
+      (oldVM: VMLikeEntityKind, newVM: VMLikeEntityKind) =>
+        _.isEqual(getTolerations(oldVM), getTolerations(newVM)),
+    );
+
+    const onTolerationAdd = () =>
+      onLabelAdd({
+        id: null,
+        key: '',
+        value: '',
+        effect: TOLERATIONS_EFFECTS[0],
+      } as TolerationLabel);
+
+    const onReload = () => {
+      reload();
+      setTolerationsLabels(
+        getTolerations(vmLikeFinal)?.map((toleration, id) => ({
+          ...toleration,
+          id,
+        })) || [],
+      );
+    };
+
+    const onSubmit = async () => {
+      const k8sTolerations = tolerationsLabels.filter(({ key }) => !!key);
+
+      if (!_.isEqual(getTolerations(vmLikeFinal), k8sTolerations)) {
+        // eslint-disable-next-line promise/catch-or-return
+        handlePromise(
+          k8sPatch(
+            getVMLikeModel(vmLikeFinal),
+            vmLikeFinal,
+            await getTolerationsPatch(vmLikeFinal, k8sTolerations),
+          ),
+        ).then(close);
+      } else {
+        close();
+      }
+    };
+
+    const extraLabelColumn = {
+      title: 'Effect',
+      column: (label: TolerationLabel) => (
+        <GridItem span={3}>
```

Comment:
hard coded span is not really extensible

Reply:
why hard coded? this is a user implementation of the extendibility. 
Its very clear, the user can add 1 extra column if he wants, with a label and a column.
the label row must have span 4,4,1 for the label, value, and delete button.
if 3 isn't enough for the dev, he should create his own row.

Reply:
because it is very hard to change the implementation in the future with these use cases .

Because the column relies on the inner workings of the row (usage of Grid) + the size as well.

plus this isn't really the react way of extending things as already said